index.bs

```diff
@@ -2156,10 +2189,121 @@ Base64-encoded string.
 
 TODO: Full page screenshots, and multiple output formats
 
+<div algorithm>
+To <dfn>normalise rect</dfn> given |rect|:
+
+Note: This ensures that the resulting rect has positive [=width dimension=] and
+[=height dimension=].
+
+1. Let |x| be |rect|'s [=x coordinate=].
+
+1. Let |y| be |rect|'s [=y coordinate=].
+
+1. Let |width| be |rect|'s [=width dimension=].
+
+1. Let |height| be |rect|'s [=height dimension=].
+
+1. If |width| is less than 0, set |x| to |x| + |width| and then set |width| to
+   -|width|.
+
+1. If |height| is less than 0, set |y| to |y| + |height| and then set |height|
+   to -|height|.
+
+1. Return a new {{DOMRectReadOnly}} with [=x coordinate=] |x|, [=y coordinate=]
+   |y|, [=width dimension=] |width| and [=height dimension=] |height|.
+
+</div>
+
+<div algorithm>
+To <dfn>rectangle intersection</dfn> given |rect1| and |rect2|
```

Comment:
Generally defining things in terms of JS is a bad idea; it's very heavyweight to call JS code from a spec context, and if you're not careful it leaves a bunch of questions unanswered (e.g. in this case, which realm are you running in? what happens if `IntersectionObserver` is redefined in that realm? What happens if the code throws an exception? Are you correctly waiting for promises to resolve? How do you read the result back into spec types?). 

For all these reasons, it's strongly preferred to call spec algorithms rather than trying to define things in terms of js. In this case the underlying spec algorithm seems to be https://w3c.github.io/IntersectionObserver/#compute-the-intersection but I think that doesn't quite fit our requirements, which really are just about figuring out the overlap of two rectangles.

Reply:
Makes sense!